From 245ee4c5e13381882c3a41607305b2d02f544b22 Mon Sep 17 00:00:00 2001
From: Ben Levinsky <ben.levinsky@amd.com>
Date: Tue, 28 Feb 2023 08:15:06 -0800
Subject: [PATCH] lib: atomic: add defines for build

symbols ATOMIC_VAR_INIT and atomic_init cause build errors

Match from upstream
https://github.com/gcc-mirror/gcc/blob/master/gcc/ginclude/stdatomic.h
To make build work

Signed-off-by: Ben Levinsky <ben.levinsky@amd.com>
---
 lib/atomic.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lib/atomic.h b/lib/atomic.h
index 44d6b9a..e641097 100644
--- a/lib/atomic.h
+++ b/lib/atomic.h
@@ -98,6 +98,14 @@ using std::atomic_signal_fence;
       !defined(__STDC_NO_ATOMICS__)
 # include <stdint.h>
 # include <stdatomic.h>
+
+#undef ATOMIC_VAR_INIT
+#define ATOMIC_VAR_INIT(VALUE)	(VALUE)
+
+#undef atomic_init
+#define atomic_init(PTR, VAL)                           \
+  atomic_store_explicit (PTR, VAL, __ATOMIC_RELAXED)
+
 #elif defined(__GNUC__)
 # include <metal/compiler/gcc/atomic.h>
 #else
-- 
2.25.1

